command/push.go

```diff
@@ -16,6 +17,11 @@ import (
 // archiveTemplateEntry is the name the template always takes within the slug.
 const archiveTemplateEntry = ".packer-template"
 
+var (
+	reName         = regexp.MustCompile("^[a-zA-Z0-9-_/]+$")
+	errInvalidName = fmt.Errorf("Your build name can only contain these characters: [a-zA-Z0-9-_]+")
```

Comment:
The error message should use `reName.String()` instead of duplicating the data and getting it wrong.